api/environments/identities/models.py

```diff
@@ -235,23 +240,30 @@ def update_traits(self, trait_data_items):
         """
         current_traits = {t.trait_key: t for t in self.identity_traits.all()}
 
-        keys_to_delete = []
+        keys_to_delete = set()
         new_traits = []
         updated_traits = []
+        transient_traits = []
 
         for trait_data_item in trait_data_items:
             trait_key = trait_data_item["trait_key"]
             trait_value = trait_data_item["trait_value"]
+            transient = trait_data_item.get("transient")
 
             if trait_value is None:
                 # build a list of trait keys to delete having been nulled by the
                 # input data
-                keys_to_delete.append(trait_key)
+                keys_to_delete.add(trait_key)
```

Comment:
This is an interesting design question (but a fairly slim use case). If a trait is marked as transient, and set to null, should it be able to delete an existing persisted trait 🤔 ?

Reply:
IMO no, it should not.

Reply:
I agree. But then I _think_ the code needs updating?

Reply:
This is done. Also added two integration tests as per the comment[here](https://github.com/Flagsmith/flagsmith/pull/4325#pullrequestreview-2175017859).